pkg/ebpf/oom_kill.go

```diff
@@ -0,0 +1,113 @@
+// +build linux_bpf,bcc
+
+package ebpf
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"regexp"
+	"unsafe"
+
+	"github.com/DataDog/datadog-agent/pkg/ebpf/bytecode"
+	"github.com/DataDog/datadog-agent/pkg/ebpf/oomkill"
+
+	bpflib "github.com/iovisor/gobpf/bcc"
+)
+
+/*
+#include <string.h>
+#include "c/oom-kill-kern-user.h"
+*/
+import "C"
+
+type OOMKillProbe struct {
+	m      *bpflib.Module
+	oomMap *bpflib.Table
+}
+
+func NewOOMKillProbe() (*OOMKillProbe, error) {
+	source_raw, err := bytecode.Asset("oom-kill-kern.c")
+	if err != nil {
+		return nil, fmt.Errorf("Couldn’t find asset “oom-kill-kern.c”: %v", err)
+	}
+
+	// Process the `#include` of embedded headers.
+	// Note that embedded headers including other embedded headers is not managed because
+	// this would also require to properly handle inclusion guards.
+	includeRegexp := regexp.MustCompile(`^\s*#\s*include\s+"(.*)"$`)
```

Comment:
But the regex is contant.
So, we definitively don’t expect a runtime panic here. Once the regex is correct, it will necessarily always compile.
We typically do here something that should have been done at compile time !